Pysa scraper crashing

**Describe the bug**
Pysa scraper produces an error: if len(published.split('/')[2]) == 4:
/ IndexError: list index out of range

**To Reproduce**
1. Input right Pysa site URL
2. Run ransomwatch task

**Expected behavior**
Scraper works on Pysa.

**Screenshots**
![image](https://user-images.githubusercontent.com/45006777/143707164-5e70e19e-5b41-4b4e-a5d4-c085780404b0.png)


**Logs**
app_1    | 2021/11/27 18:09:50 [ERROR] Got an error while scraping Pysa, notifying
app_1    | 2021/11/27 18:09:50 [ERROR] Traceback (most recent call last):
app_1    |   File "/app/ransomwatch.py", line 77, in main
app_1    |     s.scrape_victims()
app_1    |   File "/app/sites/pysa.py", line 33, in scrape_victims
app_1    |     if len(published.split('/')[2]) == 4:
app_1    | IndexError: list index out of range

**Environment**
 - OS: Linux Debian 10
 - How you are running it: Docker local image

**Additional context**
-

--- a/requirements.txt
+++ b/requirements.txt
@@ -6,3 +6,4 @@ SQLAlchemy
 pyyaml
 python-dateutil
 lxml
+dateparser

--- a/src/sites/pysa.py
+++ b/src/sites/pysa.py
@@ -1,13 +1,13 @@
-from datetime import datetime
-from dateutil.relativedelta import relativedelta
 import logging
+from datetime import datetime
 
 from bs4 import BeautifulSoup
+import dateparser
+from dateutil.relativedelta import relativedelta
 
 from db.models import Victim
 from net.proxy import Proxy
 from .sitecrawler import SiteCrawler
-
 
 class Pysa(SiteCrawler):
     actor = "Pysa"
@@ -21,34 +21,17 @@ class Pysa(SiteCrawler):
 
             # get max page number
             victim_list = soup.find_all("div", class_="page-header")
-            last_published_dt = None
 
             for victim in victim_list:
                 victim_name = victim.find_all("a")[0].text.strip()
                 published = victim.find_all("span")[1].text.strip()
 
-                # they use a bunch of different date format...
-                published_dt = None
-                year_format = "%y"
-                if len(published.split('/')[2]) == 4:
-                    year_format = "%Y"
-                try:
-                    published_dt = datetime.strptime(
-                        published, "%m/%d/{}".format(year_format))
-                    # as we don't know the format of the date, let's check if the month detected
-                    # is near the previous considered month
-                    if last_published_dt and ((published_dt.month != last_published_dt.month)
-                        or (published_dt.month != (last_published_dt + relativedelta(months=-1)).month)):
-                        published_dt = datetime.strptime(
-                            published, "%d/%m/{}".format(year_format))
-                except ValueError as ve:
-                    try:
-                        published_dt = datetime.strptime(
-                            published, "%d/%m/{}".format(year_format))
-                    except ValueError as ve:
-                        logging.error("Unable to cast date {}".format(published))
-
-                last_published_dt = published_dt
+                # they use a bunch of different date format
+                # use a nice dateparsing library to handle them all in an easier manner
+                published_dt = dateparser.parse(published)
+                # sometimes they don't have a timestamp
+                if published_dt is None and len(published) > 0:
+                    logging.warning(f"couldn't parse timestamp: {published}")
 
                 victim_leak_site = self.url + '/' + victim.find_all("a")[0].attrs["href"]
                 q = self.session.query(Victim).filter_by(

